File: OSD/Inode/inode1.py
import os
import sys,stat
filename=raw_input("Enter the name of file")
fl=os.open(filename,os.O_RDWR)
info=os.fstat(fl)

print "1. Size of file is",info.st_size
print "2. ID of device containing file ",info.st_dev
print "3. Inode number is",info.st_ino
print "4. Mode:(octal)",info.st_mode
print "5. Link count:",info.st_nlink
print "6. UID owner:",info.st_uid
print "7. GID owner:",info.st_gid
print "8. Preferred I/O block size:",info.st_blksize,"In Bytes"
print "9. File size:",info.st_size,"In Bytes"
print "10. Blocks allocated:",info.st_blocks,"In Bytes"
print "11. Time of last access:",info.st_atime
print "12. Time of last modification:",info.st_mtime
print "13. Time of status change:",info.st_ctime



if stat.S_ISREG(info.st_mode):
	print "REGULAR FILE"
elif stat.S_ISDIR(info.st_mode):
	print "DIRECTORY FILE"
elif stat.S_ISCHR(info.st_mode):
	print "CHARACTER FILE"
elif stat.S.ISBLK(info.st_mode):
	print "BLOCK FILE"
elif stat.S.ISFIFO(info.st_mode):
	print " FIFO FILE"
elif stat.S_ISLINK(info.st_mode):
	print " SYMBOLIC LINK"
elif stat.S_ISSOCK(info.st_mode):
	print " SOCKET"

File: OSD/Inode/inode1.cpp
/*To Run The Program
 Syntax :- g++ Program_name.cpp &
 	./a.out
*/
#include<iostream>
#include<stdlib.h>
#include<stdio.h>
#include<sys/stat.h>
using namespace std;
int main(int argc,char *argv[])
{
struct stat sb;
FILE *ptr;
char filename[10];
cout<<"Enter the name of file";
cin>>filename;

stat(filename,&sb);
if(stat(filename,&sb)==-1){
	cout<<"file does not exist";
	cout<<"Creating file ";
	ptr=fopen(filename,"wb");
	fclose(ptr);
	}
else   {
stat(filename,&sb);
 if(S_ISBLK(sb.st_mode)){cout<<"block device\n";}            
  else if(S_ISCHR(sb.st_mode)){cout<<"character device\n";} 
  else if(S_ISDIR(sb.st_mode)){cout<<"directory\n";}
  else if(S_ISLNK(sb.st_mode)){cout<<"symlink\n";}  
  else if(S_ISREG(sb.st_mode)){cout<<"regular file\n";}
  else if(S_ISSOCK(sb.st_mode)){cout<<"socket\n";}      
   else {cout<<"unknown?\n";}    


cout<<"\n1.The size of file is"<<sb.st_size;
cout<<"\n2.Inode Number of file is"<<sb.st_ino;
cout<<"\n3.Mode:(octal)"<<sb.st_mode;
cout<<"\n4.Link count:"<<sb.st_nlink;
cout<<"\n5.Ownership:UID="<<sb.st_uid<<"\n \tGID="<<sb.st_gid;
cout<<"\n6.Preferred I/O block size:"<<sb.st_blksize<<"In Bytes";
cout<<"\n7.File size:"<<sb.st_size<<"In Bytes";
cout<<"\n8.Blocks allocated:"<<sb.st_blocks<<"In Bytes";
cout<< "\n9. Time of last access:"<<sb.st_atime;
cout<<"\n10. Time of last modification:"<<sb.st_mtime;
cout<<"\n11. Time of status change:"<<sb.st_ctime;

}
}

